### PR TITLE
feat: added uniq hash in kubeconfig_path

### PR DESCRIPTION
**Pull request by pandwoter**
Hello this is the fix for following issue (https://github.com/getkuby/kuby-digitalocean/issues/2)
also, it depends on this PR (https://github.com/getkuby/kuby-core/pull/43)

### CODE REVIEW

**Comment by pandwoter** [comment]
I'm not really sure why but on my local setup ```config.access_token + config.cluster_id``` says that config.access_token and config.cluster_id is nil (but they are provided in kuby.rb). 

So can you please point me when actually ```configure``` method of provider is called and is there a situations when Provider is called without executing a ```configure``` method?

File: lib/kuby/digitalocean/provider.rb [hunk 1]
```diff
@@ -16,8 +16,11 @@ def configure(&block)
       end
 
       def kubeconfig_path
-        @kubeconfig_path ||= File.join(
-          kubeconfig_dir, "#{environment.app_name.downcase}-kubeconfig.yaml"
+        File.join(
+          kubeconfig_dir,
+          "#{environment.app_name.downcase}" \
+          "-#{generate_hash(config.access_token + config.cluster_id)}" \
```
Comment (camertron) [inline lib/kuby/digitalocean/provider.rb]: Can we move `generate_hash` into this class? Defining it in the superclass means 1) we'd have to make kuby-digitalocean depend on a specific version of kuby-core, and 2) we're saying the hash function needs to be the same across all providers. All we really care about is the uniqueness of the path on disk. The mechanism that provides that unique path can (and should IMHO) be provider-independent.

**Review by camertron — CHANGES_REQUESTED** [review]

**Comment by camertron** [comment]
@pandwoter hmm I'm not sure why that would happen either. How are you testing gem changes locally? The `configure` method is called [here](https://github.com/getkuby/kuby-core/blob/c56945db98a68bc07fd93e9b7b75d98ba2deae0a/lib/kuby/kubernetes/spec.rb#L37).

**Comment by pandwoter** [comment]
Done!

**Comment by camertron** [comment]
@pandwoter ok, this looks awesome! I did a little testing and encountered the same `nil`s for cluster ID and access token. It happens because, unfortunately, the `kubeconfig_path` method is called before Kuby has been fully initialized. Fortunately there's a fairly easy workaround we can use to call `kubeconfig_path` a little later in the process. I'll merge this PR then add in the fix and publish a new release.

**Comment by camertron** [comment]
Published as v0.4.3, thanks again!

**Pull request merged**